### ISSUE TITLE
transform.resize throws ValueError with np.ndarray output_shape when dim is not provided

### ISSUE THREAD

**Issue by alicanb** [scikit-image/scikit-image#6224]
## Way to reproduce
```python
from skimage.transform import resize
import numpy as np
resize(np.random.rand(100, 100, 3), np.array((30, 30,3))) # works
resize(np.random.rand(100, 100, 3), (30, 30)) # works
resize(np.random.rand(100, 100, 3), np.array((30, 30))) # ValueError: operands could not be broadcast together with shapes (3,) (2,) 
```

## Version information
```python
# Paste the output of the following python commands
from __future__ import print_function
import sys; print(sys.version)
import platform; print(platform.platform())
import skimage; print(f'scikit-image version: {skimage.__version__}')
import numpy; print(f'numpy version: {numpy.__version__}')
```

```python
3.8.10 (default, Nov 26 2021, 20:14:08) 
[GCC 9.3.0]
Linux-5.10.76-linuxkit-x86_64-with-glibc2.29
scikit-image version: 0.19.1
numpy version: 1.21.5
```

**Comment by grlee77** [comment]
Thanks @alicanb, that is odd behavior. I was planning to try and release a new patch release soon. Let me add this to the milestone

**Comment by grlee77** [comment]
I checked on the `main` branch and this issue does not occur there. I was able to reproduce it on the `v0.19.x` branch. I will try to backport a fix to `v0.19.x` now.

**Comment by grlee77** [comment]
closed by #6222
